### PR TITLE
feature(nemesis.py): add new scrub nemesis

### PR DESCRIPTION
**Pull request by amoskong**
Signed-off-by: Amos Kong <amos@scylladb.com>

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I gave variables/functions meaningful self-explanatory names
- [ ] I didn't leave commented-out/debugging code
- [ ] I didn't copy-paste code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by roydahan** [comment]
Is it ready for review or still draft?
Looks good to me assuming it's tested on real longevity and works...

One thing we need to take into account.
It's a nemesis that is supported only on 3.3 / 2019.1.6 and above, right?
It means SCT won't be able to use older versions.

@bentsi maybe we should add a "since" decorator for nemesis.

**Comment by bentsi** [comment]
@roydahan we are aware of the need to support "@since" functionality and it is already in the https://trello.com/c/sIpowINg/683-redesign-nemesis-framework task
we can make a quick-n-dirty solution for current Nemesis

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -366,7 +366,10 @@ def _destroy_data_and_restart_scylla(self):
             for _ in range(5):
                 file_for_destroy = self._choose_file_for_destroy(ks_cfs)
 
-                result = self.target_node.remoter.run('sudo rm -f %s' % file_for_destroy)
+                if corrupt_by_dd:
+                    result = self.target_node.remoter.run('sudo dd if=/dev/random of=%s count=1024' % file_for_destroy)
```
Comment (bentsi) [inline sdcm/nemesis.py]: better explicitly mention block size (bs)

Comment (bentsi) [reply sdcm/nemesis.py]: question: why did you choose to create a 512k file?

Comment (roydahan) [reply sdcm/nemesis.py]: He just write some bytes into an sstable in order to corrupt it.

**Comment by bentsi** [comment]
@amoskong please resolve conflicts

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -1817,6 +1822,17 @@ def disrupt_repair_streaming_err(self):
...
         self.break_streaming_task_and_rebuild(task='repair')
 
+    def disrupt_scrub(self):
+        self.log.debug("Corrupt the sstable file")
+        self._destroy_data_and_restart_scylla(corrupt_by_dd=True)
+
+        self.log.debug("Rebuild sstables by sstablescrub")
+        for ks_cf in get_non_system_ks_cf_list(self.loaders.nodes[0], self.cluster.nodes[0]):
+            scrub_cmd = 'curl -s -X GET --header "Content-Type: application/json" --header ' \
+                        '"Accept: application/json" "http://127.0.0.1:10000/storage_service/keyspace_scrub/{}?skip_corrupted=true"'.format(
```
Comment (roydahan) [inline sdcm/nemesis.py]: Open yourself a task to go back here and change it to nodetool scrub when the issue for nodetool is fixed.

Comment (roydahan) [reply sdcm/nemesis.py]: Also for the dtest.

Comment (amoskong) [reply sdcm/nemesis.py]: Opened a ticket: https://trello.com/c/jBSfwaMZ/1817-switch-to-use-nodetool-scrub-from-api-sct-dtest

**Comment by roydahan** [comment]
need to fix the failed pre-commit so we can merge.

**Comment by bentsi** [comment]
what about backport labels?

**Comment by amoskong** [comment]
> what about backport labels?

It's required for branch-2019.1, but let's run the nemesis for a while, if it's stable I will send a PR to backport all patches (PRs) once to branch-2019.1.
@bentsi Is it ok?

**Pull request merged**